kube-core/src/dynamic.rs

```diff
@@ -1,14 +1,28 @@
 //! Contains types for using resource kinds not known at compile-time.
 //!
 //! For concrete usage see [examples prefixed with dynamic_](https://github.com/kube-rs/kube/tree/main/examples).
-
 pub use crate::discovery::ApiResource;
 use crate::{
     metadata::TypeMeta,
     resource::{DynamicResourceScope, Resource},
 };
+
 use k8s_openapi::apimachinery::pkg::apis::meta::v1::ObjectMeta;
 use std::borrow::Cow;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+#[error("failed to parse this DynamicObject into a Resource")]
+/// Failed to parse `DynamicObject` into `Resource`
+pub struct ParseDynamicObjectError {
+    source: serde_json::Error,
```

Comment:
IIRC we can do a [`#[from]` attr](https://docs.rs/thiserror/latest/thiserror/index.html#details) here to avoid the manual From impl.

Normally we do these error types with enums, but I think this is OK given the single entry.

Reply:
You recalled correctly, awesome stuff.